src/dmd/backend/cc.d

```diff
@@ -1436,7 +1436,14 @@ struct Symbol
     uint lnoscopestart;         // life time of var
     uint lnoscopeend;           // the line after the scope
 
-    char[1] Sident;             // identifier string (dynamic array)
+    /**
+     * Identifier for this symbol
+     *
+     * Note that this is used as a flexible array member.
+     * When allocating a Symbol, the allocation is for
+     * `sizeof(Symbol - 1 + strlen(identifier) + "\0".length)`.
+     */
+    char[1] Sident;
```

Comment:
Isn't `char[0]` suppose to be use for that?

Reply:
I didn't write that code, I am merely documented its usage 🤷‍♂